crates/houston/src/profile/mod.rs

```diff
@@ -121,3 +121,35 @@ impl fmt::Display for Profile {
         write!(f, "{}", self.sensitive)
     }
 }
+
+// Masks all but the first 4 and last 4 chars of a key with a set number of *
+// valid keys are all at least 22 chars. We don't care if invalid keys
+// are printed, so we don't need to worry about strings 8 chars or less,
+// which this fn would just print back out
+pub fn mask_key(key: &Option<String>) -> Option<String> {
+    if let Some(key) = key {
+        let ex = regex::Regex::new(r"(?im)^(.{4})(.*)(.{4})$").unwrap();
+        let masked = ex.replace(key, "$1******************$3").into_owned();
+        Some(masked)
+    } else {
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::mask_key;
+    #[test]
```

Comment:
To [make clippy happy](https://rust-lang.github.io/rust-clippy/master/#many_single_char_names) here:
```suggestion
    #[allow(clippy::many_single_char_names)]
    #[test]
```